ownCloudAppShared/Tools/AppStatistics.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  AppStatistics.swift
+//  ownCloudAppShared
+//
+//  Created by Michael Neuwert on 10.06.20.
+//  Copyright © 2020 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2020, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import Foundation
+import ownCloudSDK
+import StoreKit
+
+extension TimeInterval {
+    public var days: Int {
+        return Int(self / (3600.0 * 24.0))
+    }
+}
+
+public class AppStatistics {
+
+    enum UserDefaultsKeys: String {
+        case launchDateDefaultsKey, updateDateDefaultsKey, lastVersionDefaultsKey, lastPromptedAboutReviewKey
+    }
+
+    private var launchDate: Date
+    private var updateDate: Date
+
+    private var lastReviewPromptDate: Date? {
+        get {
+            return OCAppIdentity.shared.userDefaults?.object(forKey: UserDefaultsKeys.lastPromptedAboutReviewKey.rawValue) as? Date
+        }
+
+        set {
+            OCAppIdentity.shared.userDefaults?.set(newValue, forKey: UserDefaultsKeys.lastPromptedAboutReviewKey.rawValue)
+            OCAppIdentity.shared.userDefaults?.synchronize()
+        }
+    }
+
+    public static let shared = AppStatistics()
+
+    public var timeIntervalSinceFirstLaunch: TimeInterval {
+        return Date().timeIntervalSince(launchDate)
+    }
+
+    public var timeIntervalSinceUpdate: TimeInterval {
+        return Date().timeIntervalSince(updateDate)
+    }
+
+    private var appVersion: String? {
+        guard let version = OCAppIdentity.shared.appVersion, let build = OCAppIdentity.shared.appBuildNumber else {
+            return nil
+        }
+        return "\(version).\(build)"
+    }
+
+    init() {
+        let now = Date()
+        launchDate = now
+        updateDate = now
```

Comment:
I believe `update()` should be called here, too. Otherwise `timeIntervalSinceFirstLaunch` and `timeIntervalSinceUpdate` will return wrong values.

Reply:
@felix-schwarz It is called in AppDelegate: `AppStatistics.shared.update()` in `didFinishLaunching` method. This is an intention, otherwise it would a bit uglier construct IMHO` let _ = AppStatistics.shared()`

Reply:
@felix-schwarz fixed